drivers/docker/driver.go

```diff
@@ -0,0 +1,1682 @@
+		NetworkingConfig: networkingConfig,
+	}, nil
+}
+
+// detectIP of Docker container. Returns the first IP found as well as true if
+// the IP should be advertised (bridge network IPs return false). Returns an
+// empty string and false if no IP could be found.
+func (d *Driver) detectIP(c *docker.Container, driverConfig *TaskConfig) (string, bool) {
+	if c.NetworkSettings == nil {
+		// This should only happen if there's been a coding error (such
+		// as not calling InspectContainer after CreateContainer). Code
+		// defensively in case the Docker API changes subtly.
+		d.logger.Error("no network settings for container", "container_id", c.ID)
+		return "", false
+	}
+
+	ip, ipName := "", ""
+	auto := false
+	for name, net := range c.NetworkSettings.Networks {
+		if net.IPAddress == "" {
+			// Ignore networks without an IP address
+			continue
+		}
+
+		ip = net.IPAddress
+		if driverConfig.AdvertiseIPv6Addr {
+			ip = net.GlobalIPv6Address
+			auto = true
+		}
+		ipName = name
+
+		// Don't auto-advertise IPs for default networks (bridge on
+		// Linux, nat on Windows)
+		if name != "bridge" && name != "nat" {
+			auto = true
+		}
+
+		break
+	}
+
+	if n := len(c.NetworkSettings.Networks); n > 1 {
+		d.logger.Warn("multiple Docker networks for container found but Nomad only supports 1",
+			"total_networks", n,
+			"container_id", c.ID,
+			"container_network", ipName)
+	}
+
+	return ip, auto
+}
+
+// validateCommand validates that the command only has a single value and
+// returns a user friendly error message telling them to use the passed
+// argField.
+func validateCommand(command, argField string) error {
+	trimmed := strings.TrimSpace(command)
+	if len(trimmed) == 0 {
+		return fmt.Errorf("command empty: %q", command)
+	}
+
+	if len(trimmed) != len(command) {
+		return fmt.Errorf("command contains extra white space: %q", command)
+	}
+
+	return nil
+}
+
+func (d *Driver) WaitTask(ctx context.Context, taskID string) (<-chan *drivers.ExitResult, error) {
+	h, ok := d.tasks.Get(taskID)
+	if !ok {
+		return nil, drivers.ErrTaskNotFound
+	}
+	ch := make(chan *drivers.ExitResult)
+	go d.handleWait(ctx, ch, h)
+	return ch, nil
+}
+
+func (d *Driver) handleWait(ctx context.Context, ch chan *drivers.ExitResult, h *taskHandle) {
+	defer close(ch)
+	select {
+	case <-h.waitCh:
+		ch <- h.exitResult
+	case <-ctx.Done():
+		ch <- &drivers.ExitResult{
+			Err: ctx.Err(),
+		}
+	}
+}
+
+func (d *Driver) StopTask(taskID string, timeout time.Duration, signal string) error {
+	h, ok := d.tasks.Get(taskID)
+	if !ok {
+		return drivers.ErrTaskNotFound
+	}
+
+	if signal == "" {
+		signal = "SIGINT"
+	}
+
+	sig, err := signals.Parse(signal)
+	if err != nil {
+		return fmt.Errorf("failed to parse signal: %v", err)
+	}
+
+	return h.Kill(timeout, sig)
+}
+
+func (d *Driver) DestroyTask(taskID string, force bool) error {
+	h, ok := d.tasks.Get(taskID)
+	if !ok {
+		return drivers.ErrTaskNotFound
+	}
+
+	defer h.dloggerPluginClient.Kill()
```

Comment:
Move _after_ Running+force check. Otherwise you'll kill the logger when force=false and the container is running.

Nothing past that point `return`s early anyway so you can drop the defer and put this last.